packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -106,15 +106,49 @@ function hydrateNode(node: Node, vnode: VNode, renderer: RendererAPI): Node | nu
 }
 
 const NODE_VALUE_PROP = 'nodeValue';
+const PARENT_NODE_PROP = 'parentNode';
+const TAG_NAME_PROP = 'tagName';
+
+function textNodeContentsAreEqual(node: Node, vnode: VText, renderer: RendererAPI): boolean {
+    const { getProperty } = renderer;
+    const nodeValue = getProperty(node, NODE_VALUE_PROP);
+
+    if (nodeValue === vnode.text) {
+        return true;
+    }
+
+    // Special case for empty text nodes – these are serialized differently on the server
+    if (nodeValue === '\u200D' && vnode.text === '') {
```

Comment:
I had to go back to the [original PR](https://github.com/salesforce/lwc/pull/2656) to remember why this is true. Thanks for adding context here.

Reply:
Thanks for the link! Added it to the comment.